iotdb-core/datanode/src/main/java/org/apache/iotdb/db/consensus/statemachine/dataregion/DataRegionStateMachine.java

```diff
@@ -287,15 +287,16 @@ public DataSet read(IConsensusRequest request) {
 
   @Override
   public File getSnapshotRoot() {
-    String snapshotDir =
-        IoTDBDescriptor.getInstance().getConfig().getRatisDataRegionSnapshotDir()
-            + File.separator
-            + region.getDatabaseName()
-            + "-"
-            + region.getDataRegionId();
+    String snapshotDir = "null";
     try {
+      snapshotDir =
+          IoTDBDescriptor.getInstance().getConfig().getRatisDataRegionSnapshotDir()
+              + File.separator
+              + region.getDatabaseName()
+              + "-"
+              + region.getDataRegionId();
       return new File(snapshotDir).getCanonicalFile();
-    } catch (IOException e) {
+    } catch (IOException | NullPointerException e) {
```

Comment:
why catch NullPointerException?

Reply:
Because we found there may be a NPE in IT test, and this function could return null, so we catch here and return null.

Reply:
String snapshotDir = "null"; just want to distinguish whether it is a npe or a file exception. It can alse change to "".

Reply:
can also spend time considering other solutions ...